Clarify pull request approval by engineering (#524)
GSA-TTS/identity-handbook: _articles/pull-request-review.md

[github-repositories]: {% link _articles/github.md %}#repositories

> - **Who merges**: once there is at least one approving review by a member of
>   the engineering team, the author merges their own pull request.

Reviewer: what if we broke out "who reviews" into its own bullet, since this is the "reviewing and merging" section?

```suggestion
- **Who reviews**: at least one member of the engineering must review and approve a pull request before it can be merged
- **Who merges**: the author merges their own pull request.
```

Author: Hmm, now that you mention it, there already is a section below for "Who reviews" and it does touch on this aspect of needing engineering review, though indirectly through GitHub groups, and in language which leaves room for ambiguity ("in general [...] is adequate").

I'll try to come up with a revision that expands within the existing "Who reviews"

Author: Updated in 8435cb5:

- Moved "Who reviews" to come first
- Include language "at least one member of the engineering team must review and approve" in "Who reviews"
- Move existing "in general [...] is adequate" language to sub-bullet for more detail
- Remove redundant "after X reviews" language in "Who merges"
